sql/08-is_table_versioned.sql

```diff
@@ -1,28 +1,72 @@
 
+CREATE OR REPLACE FUNCTION _ver_is_table_versioned(
+    p_schema NAME,
+    p_table  NAME
+)
+RETURNS BOOLEAN AS
+$$
+    SELECT EXISTS (
+      SELECT
+          versioned
+      FROM
+          @extschema@.versioned_tables
+      WHERE
+          schema_name = p_schema AND
+          table_name = p_table
+    )
+$$ LANGUAGE sql STABLE;
+
+-- {
 CREATE OR REPLACE FUNCTION ver_is_table_versioned(
     p_schema NAME,
     p_table  NAME
 )
-RETURNS BOOLEAN AS 
+RETURNS BOOLEAN AS
 $$
 DECLARE
     v_is_versioned BOOLEAN;
+    v_has_trigger BOOLEAN;
 BEGIN
     SELECT
         versioned
     INTO
         v_is_versioned
-    FROM 
-        @extschema@.versioned_tables 
+    FROM
+        @extschema@.versioned_tables
     WHERE
         schema_name = p_schema AND
         table_name = p_table;
 
     IF v_is_versioned IS NULL THEN
-        v_is_versioned := FALSE;
+        RETURN FALSE;
+    END IF;
+
+    -- Check that table exists and triggers are in place;
+    -- warn otherwise, see:
+    -- https://github.com/linz/postgresql-tableversion/issues/57
+    BEGIN
+      v_is_versioned := EXISTS (
+        SELECT *
+      FROM pg_trigger t
+      WHERE t.tgrelid = (quote_ident(p_schema) || '.' || quote_ident(p_table) )::regclass
+        AND t.tgname = p_schema || '_' || p_table || '_revision_trg'
+      );
+    EXCEPTION
```

Comment:
Should this exception block not be around "FROM @extschema@.versioned_tables" statement?

Reply:
That block is not there anymore now that we use the internal function.
Anyway I don't think we want to catch an exception from lack of versioned_table, if that table is missing it's ok to throw an exception

Reply:
yes, it's to catch ::regclass cast